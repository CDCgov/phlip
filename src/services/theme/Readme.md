The project uses the material-ui library for React. Material-UI is based
on Google's Material Design. There's a global application theme of colors
and overrides that is given to the `<MuiThemeProvider>` component in
`src/App.js`.

To create the theme, the `createMuiTheme` function from the library is
used. We pass in the primary, secondary and error colors to the palette
property that can be referenced throughout the app on Material UI components.

Material UI allows you to override any default configuration. This project
has four overrides, defined in the `overrides` property:

- Form labels are set to the secondary color (teal) when focused
- Form labels are set to a gray color when disabled
- The underline of a text field when focus is the secondary color (teal)
- Avatar images are set to 100% width and 100% height, otherwise images
are made to fit the circle and may not fill the entire circle
