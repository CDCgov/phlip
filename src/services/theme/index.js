import { createMuiTheme } from '@material-ui/core/styles'
import { pink, deepPurple, teal } from '@material-ui/core/colors'

const mainColor = '#048484'

const theme = createMuiTheme({
  palette: {
    primary: {
      ...deepPurple,
      main: '#3d316a',
      light: '#6a5b98',
      dark: '#100b3f',
      500: '#3d316a'
    },
    secondary: {
      ...teal,
      main: mainColor,
      light: '#4fb4b4',
      dark: '#005757',
      pageHeader: '#005757',
      'A100': mainColor,
      'A200': mainColor, // MUI uses this color to determine color of buttons when set to 'secondary' (beta-25)
      'A400': '#005757', // MUI uses this color to determine color of buttons on hover when set to 'secondary' (beta-25)
      tabs: '#ecf7f6'
    },
    error: {
      ...pink,
      500: '#db036e',
      main: '#db036e'
    },
    greyText: '#757575'
  },
  scenes: {
    background: '#f5f5f5'
  },
  shape: {
    borderRadius: 2
  },
  overrides: {
    MuiFormLabel: {
      root: {
        '&$focused': {
          color: mainColor
        },
        '&$disabled': {
          color: 'black'
        }
      },
    },
    MuiInput: {
      underline: {
        '&:after': {
          borderBottom: `2px solid ${mainColor}`
        }
      }
    },
    MuiAvatar: {
      img: {
        width: '100%',
        height: '100%',
        textAlign: 'center',
        maxWidth: '100%'
      }
    }
  }
})

export default theme
