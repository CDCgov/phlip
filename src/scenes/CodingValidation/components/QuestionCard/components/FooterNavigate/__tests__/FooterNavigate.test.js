import React from 'react'
import { shallow } from 'enzyme'
import { FooterNavigate } from '../index'
import theme from 'services/theme'

const props = {
  currentIndex: 0,
  getNextQuestion: () => {},
  getPrevQuestion: () => {},
  totalLength: 1,
  theme
}

describe('CodingValidation -- FooterNavigate component', () => {
  test('should render correctly', () => {
    expect(shallow(<FooterNavigate {...props} />)).toMatchSnapshot()
  })
})
