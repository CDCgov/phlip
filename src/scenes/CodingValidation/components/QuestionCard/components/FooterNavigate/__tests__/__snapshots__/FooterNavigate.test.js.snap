// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coding -- FooterNavigate component should render correctly 1`] = `
<Row
  component={
    <WithStyles(Grid)
      item={true}
    />
  }
  displayFlex={true}
  flex={false}
  reverse={false}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-end",
      "height": 50,
      "justifyContent": "space-between",
      "paddingTop": 10,
    }
  }
/>
`;
